packages/powersync_attachments_stream/lib/src/abstractions/remote_storage.dart

```diff
@@ -0,0 +1,26 @@
+import 'dart:async';
+ 
+import '../attachment.dart';
+
+/// Adapter for interfacing with remote attachment storage.
+abstract class AbstractRemoteStorageAdapter {
```

Comment:
Let's make this an interface class and just name it `RemoteStorageAdapter`:

```suggestion
abstract interface class RemoteStorageAdapter {
```